examples/tutorials/deno_doc.md

```diff
@@ -0,0 +1,415 @@
+---
+title: "Generating documentation with deno doc"
+description: "Learn how to generate professional documentation for your Deno projects using the built-in deno doc command. This tutorial covers JSDoc comments, HTML output, linting, and best practices for documenting your code."
+url: /examples/deno_doc_tutorial/
+---
+
+Good documentation is essential for any software project. It helps other
+developers understand your code, makes maintenance easier, and improves the
+overall quality of your project. Deno includes a built-in documentation
+generator called `deno doc` that can automatically generate searchable
+documentation from your TypeScript and JavaScript code.
+
+`deno doc` works out of the box, with no setup required and can generate HTML,
+JSON or terminal output. It leverages JSDoc comments and can automatically
+extract type information from your code.
```

Comment:
maybe mention JSDoc and TypeScript types, as some people will think that this will mean type information from the jsdoc comments (which we currently do not support)